language/extensions/move-table-extension/tests/TableTests.move

```diff
@@ -41,7 +41,7 @@ module Extensions::TableTests {
     }
 
     #[test]
-    #[expected_failure(abort_code = 51155)]
+    #[expected_failure(abort_code = 1)]
```

Comment:
This kind of shows why having a user level abortion code is not the best idea. This clashes with the assert code used below.

Reply:
I was ready to make a native function to expose a error code.. but it felt not quiet right..

To be fair a data structure implemented in pure Move will need to use an error like this, right?
Also to be fair `Errors::invalid_state(1)` is `257`,  `Errors::invalid_state(10)` is `2566`, and `Errors::already_published(1)` is `262` .. `Error::invalid_state(0)` is just an unfortunate combination..


Maybe it's more of the framework /stdlib's responsibility to provide a better `Error` facility? (maybe just offset everything by 10000)
Shall we just make ENOT_EMPTY more unique in this case?

Reply:
Good point. I forgot about the practice to use categories (in the first byte of the code) to provide some kind of canonical error code, and distinguish it from the reason (in your case `0`) -- a thing I pushed for myself a while ago. I think most of our native function implementations do the same and do not respect the category/reason convention.

Categories are good, but what bothers me with our current setup is that reasons are not unique across modules. That is why I played with the crypto approach: actually one can generate a (sufficiently) unique code by sha256 of the full module name and a local counter, without the need of some global error registry or convention. But perhaps we should roll out something like this systematically across all code (native and not) only once we had more sorough design discussion about this.

Can I then ask to implement for now the category/reason thing consistently for all table operations, whether native or not? That is we should also change the other abort codes in `lib.rs` to follow the same convention like `Errors.move`.

If this is PR is blocking more important work, we can also do this after it is merged, so I leave you an approve to move forward either way you prefer.

Reply:
I pushed another commit. Lmk if we want to list all error codes on both sides although not used on either side (as done in the diff), or if we want to be lean.